src/canvas/Canvas-events.spec.ts

```diff
@@ -0,0 +1,1908 @@
+
+    a.enterEditing();
+    expect(a.isEditing, 'a should be editing').toBeTruthy();
+
+    b.enterEditing();
+    expect(b.isEditing, 'b should be editing').toBeTruthy();
+    expect(a.isEditing, 'a should have exited editing').toBeFalsy();
+
+    manager.register(b);
+    manager.exitTextEditing();
+    expect(getTarget(), 'should unregister b').toBeFalsy();
+    expect(b.isEditing, 'b should exit editing').toBeFalsy();
+
+    b.enterEditing();
+    b.fire('mousedown', { e, pointer: new Point(30, 40) } as any);
+
+    expect(getTarget(), 'should register b').toBe(b);
+
+    const called: IText[] = [];
+    a.updateSelectionOnMouseMove = () => called.push(a);
+    b.updateSelectionOnMouseMove = () => called.push(b);
+
+    testCanvas.__onMouseMove(e);
+    expect(called, 'manager is called from mouse move').toEqual([b]);
+
+    manager.unregister(a);
+    expect(getTarget(), 'should not unregister b').toBe(b);
+
+    a.fire('mouseup', { e, pointer: new Point(30, 40) } as any);
+
+    expect(getTarget(), 'should not unregister b').toBe(b);
+
+    testCanvas.__onMouseUp(e);
+    expect(getTarget(), 'should unregister b').toBeFalsy();
+
+    manager.register(a);
+    expect(getTarget(), 'should register a').toBe(a);
+
+    testCanvas.remove(a);
+    expect(getTarget(), 'should unregister a').toBeFalsy();
+
+    manager.clear();
+    expect(getTarget(), 'should have disposed ref').toBeFalsy();
+    expect(getTargets(), 'should have disposed refs').toEqual([]);
+
+    const g = new Group([a]);
+    testCanvas.add(g);
+    expect(getTargets(), 'should register an existing nested instance').toEqual(
+      [a],
+    );
+
+    g.add(b);
+    expect(getTargets(), 'should register an added nested instance').toEqual([
+      a,
+      b,
+    ]);
+
+    manager.register(b);
+    expect(getTarget(), 'should register b').toBe(b);
+
+    g.remove(b);
+    expect(getTarget(), 'should unregister b').toBeFalsy();
+    expect(
+      getTargets(),
+      'should unregister a nested instance upon removal',
+    ).toEqual([a]);
+
+    manager.register(a);
+    expect(getTarget(), 'should register a').toBe(a);
+
+    testCanvas.remove(g);
+    expect(getTarget(), 'should unregister a').toBeFalsy();
+    expect(
+      getTargets(),
+      'should unregister nested instances upon group removal',
+    ).toEqual([]);
+
+    testCanvas.add(b);
+    expect(getTargets(), 'should register instance').toEqual([b]);
+
+    testCanvas.clear();
+    expect(getTargets(), 'clear should clear instances').toEqual([]);
+
+    testCanvas.add(b);
+    expect(getTargets(), 'should register instance').toEqual([b]);
+
+    manager.register(b);
+    expect(getTarget(), 'should register b').toBe(b);
+
+    await testCanvas.dispose();
+    expect(getTargets(), 'dispose should clear instances').toEqual([]);
+    expect(getTarget(), 'should unregister b').toBeFalsy();
+  });
+
+  it.todo('mousemove: subTargetCheck: setCursorFromEvent considers subTargets');
```

Comment:
I also ported these two as test todos so it doesn't get lost in case it is something that should be actually done